cli/src/main/java/com/devonfw/tools/ide/url/updater/AbstractUrlUpdater.java

```diff
@@ -551,11 +586,54 @@ private void doUpdateStatusJson(boolean success, int statusCode, String edition,
       }
       logger.warn("For tool {} and version {} the download verification failed with status code {} for URL {}.", tool,
           version, code, url);
+
       getUrlUpdaterReport().incrementVerificationFailure();
     }
     if (modified) {
       urlStatusFile.setStatusJson(statusJson); // hack to set modified (better solution welcome)
     }
+    if (status != null) {
+      assert !status.checkEmpty() : "Invalid status!";
+    }
+  }
+
+  private static void removeUrl(String url, UrlDownloadFile downloadFile, String tool, String version, Integer code, UrlStatusFile urlStatusFile,
+      UrlStatus status) {
+    logger.warn("For tool {} and version {} the the URL {} is broken (status code {}) for a long time and will be removed.", tool,
+        version, code, url);
+    downloadFile.removeUrl(url);
+    if (downloadFile.getUrls().isEmpty()) {
+      Path downloadPath = downloadFile.getPath();
+      logger.warn("For tool {} and version {} all URLs have been removed so the download file {} will be removed.", tool,
+          version, downloadPath);
+      downloadFile.delete();
+      UrlChecksum urlChecksum = downloadFile.getParent().getChecksum(downloadFile.getName());
+      if (urlChecksum == null) {
+        logger.warn("Was missing checksum file for {}", downloadFile.getPath());
+      } else {
+        urlChecksum.delete();
+      }
+    } else {
+      downloadFile.save();
+    }
+    StatusJson statusJson = urlStatusFile.getStatusJson();
+    statusJson.remove(url);
+    if (statusJson.getUrls().isEmpty()) {
+      urlStatusFile.delete();
+    } else {
+      urlStatusFile.setStatusJson(statusJson);
+    }
+  }
+
+  private static boolean isErrorCodeForAutomaticUrlRemoval(Integer code) {
+    return Integer.valueOf(404).equals(code);
```

Comment:
here we could include even more HTTP status codes like `500` or `410`.
However, this one is the really relevant one...

Reply:
In the raw logs, the only other status code for an error was `503`. So `404` should be enough.

Reply:
Thanks for the feedback.
We can still extend this here after the merge if we face new situations.
Hence resolving this.